fission-web-server/library/Fission/Web/Server/Handler/App.hs

```diff
@@ -29,8 +36,54 @@ handler ::
   , App.Retriever         t
   , App.Domain.Retriever  t
   )
-  => ServerT API.App m
-handler = Index.index
-     :<|> Create.create
-     :<|> Update.update
-     :<|> Destroy.handler
+  => App.Routes (AsServerT m)
+handler = App.Routes { destroy = genericServerT Destroy.handler, ..}
+  where
+    index Authorization {about = Entity userId _} = runDB do
+      apps        <- App.ownedBy userId
+      appXDomains <- forM apps findDomains
+      return $ Map.fromList appXDomains
+
+    create mayAppName Authorization {about = Entity userId _} = do
+      now            <- currentTime
+      (_, subdomain) <- Web.Error.ensureM $ App.createWithPlaceholder userId maySubdomain now
+      defaultDomain  <- App.Domain.initial
+
+      return URL
+        { domainName = defaultDomain
+        , subdomain  = Just subdomain
+        }
+
+      where
+        maySubdomain :: Maybe Subdomain
+        maySubdomain = App.Name.toSubdomain <$> mayAppName
+
+    update url newCID copyDataFlag Authorization {about = Entity userId _} = do
+      now <- currentTime
+      Web.Error.ensureM $ App.setCID userId url newCID copyFiles now
+      return ()
+
+      where
+        copyFiles :: Bool
+        copyFiles = maybe True identity copyDataFlag
```

Comment:
I'm *so happy* this is all inlined now. No more bazillion files.

Reply:
💯 🎉